### ISSUE TITLE
Use case for multiple calls to `apex.amp.initialize`

### ISSUE THREAD

**Issue by 7iW** [NVIDIA/apex#815]
Hello,

The documentation for the `apex.amp.initialize` function states, "If you think your use case requires `amp.initialize` to be called more than once, let us know."

My use case involves a neural architecture search technique called PC-DARTS, detailed in [this paper](https://openreview.net/forum?id=BJlS634tPr) and implemented in [this repository](https://github.com/7iW/PC-DARTS). The implementation uses two optimizers and results in the dynamic creation of many models (one new model is created for each batch of training data that is loaded).

In particular, running `python train_search.py --unrolled` on the `train_search.py` file found in the aforementioned repository results in the following situation:

- Two optimizers are instantiated before training begins, in [train_search.py#L81](https://github.com/7iW/PC-DARTS/blob/2f6aac375ada8aca3b9cbf782e456f8ce7e0243a/train_search.py#L81) (`torch.optim.SGD`) and in [architect.py#L17](https://github.com/7iW/PC-DARTS/blob/2f6aac375ada8aca3b9cbf782e456f8ce7e0243a/architect.py#L17) (`torch.optim.Adam`).
- A "main" model is instantiated before training begins, in [train_search.py#L77](https://github.com/7iW/PC-DARTS/blob/2f6aac375ada8aca3b9cbf782e456f8ce7e0243a/train_search.py#L77).
- For each batch of data loaded during the training loop:
  - A new model is created at [architect.py#L62](https://github.com/7iW/PC-DARTS/blob/2f6aac375ada8aca3b9cbf782e456f8ce7e0243a/architect.py#L62); it is instantiated with weights similar to (but not identical to) those used in the main model.
  - a loss is calculated using this newly created model; backpropagation occurs at [architect.py#L47](https://github.com/7iW/PC-DARTS/blob/2f6aac375ada8aca3b9cbf782e456f8ce7e0243a/architect.py#L47) and an optimizer step is taken at [architect.py#L37](https://github.com/7iW/PC-DARTS/blob/2f6aac375ada8aca3b9cbf782e456f8ce7e0243a/architect.py#L37) using the Adam optimizer. The newly created model is discarded shortly after backpropagation occurs.
  - a loss is calculated using the "main" model; backpropagation occurs at [train_search.py#L163](https://github.com/7iW/PC-DARTS/blob/2f6aac375ada8aca3b9cbf782e456f8ce7e0243a/train_search.py#L163) and an optimizer step is taken at [train_search.py#L165](https://github.com/7iW/PC-DARTS/blob/2f6aac375ada8aca3b9cbf782e456f8ce7e0243a/train_search.py#L165) using the SGD optimizer.

 If possible, I would like to use mixed precision for both optimizers.
